src/Components/Recommendations/BookRec.js

```diff
@@ -0,0 +1,155 @@
+import React, { useState, useEffect } from 'react';
```

Comment:
Beautiful work Ryan, this looks so clean! As I mentioned earlier if we have time we can definitely refine this function for efficiency, but it works wonderfully for now, thank you for your hard work!!

The 'RecFilter' function you wrote is actually useful in a couple places, so rather than embedding it within a react component, can you edit it to return the final list of books and export it from this file like you did the getLocalStorage function? That way we can import it to the quiz and the book page and then display the books as needed. You don't need to edit BookPage to add the import and display, I'll let the next person know how to add the function and carousels as needed!